etc/notebooks/got_scotch_demo/scotch_dashboard_declarative.ipynb

```diff
@@ -0,0 +1,783 @@
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "urth": {
+     "dashboard": {
+      "hidden": true
+     }
+    }
+   },
+   "source": [
+    "But since urth-viz-table needs the index (distillery) as a regular column with the data, we need to add that column."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "collapsed": true,
+    "urth": {
+     "dashboard": {}
+    }
+   },
+   "outputs": [],
+   "source": [
+    "def get_similar_for_table(name, n=6):\n",
+    "    # select n similar scotches except the first which is the same as name\n",
+    "    simTable = get_similar(name, n).iloc[1:]\n",
+    "    # turn the index (distillery) into a regular column\n",
+    "    return simTable.reset_index()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "urth": {
+     "dashboard": {
+      "hidden": true
+     }
+    }
+   },
+   "source": [
+    "We then need another function to retrieve the feature data for a set of scotches as *feature_keys*.\n",
+    "\n",
+    "And because urth-viz-* requires the data series as columns as opposed to rows, we will have to transpose this dataframe.\n",
+    "\n",
+    "Then similarly to the similar table above, we also need to add the index (feature) as a regular column."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "collapsed": true,
+    "urth": {
+     "dashboard": {}
+    }
+   },
+   "outputs": [],
+   "source": [
+    "def get_features_for_chart(feature_keys=[]):\n",
+    "    # select the features of the given scotches\n",
+    "    featureTable = features_df.loc[feature_keys]\n",
+    "    # transpose the dataframe and reset the index so that the first column is every feature (smokey, honey, ...)\n",
+    "    # and all the other columns are the series (scotch / distillery names)\n",
+    "    return featureTable.T.reset_index()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "urth": {
+     "dashboard": {
+      "hidden": true
+     }
+    }
+   },
+   "source": [
+    "We also need another function that takes a scotch name, a similar scotch name and returns a list of these two names. Scotch is the name user picks from the dropdown, and similar is one user picks from the suggestion table."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "collapsed": false,
+    "urth": {
+     "dashboard": {}
+    }
+   },
+   "outputs": [],
+   "source": [
+    "def get_feature_keys(scotch, similar=[]):\n",
+    "    if not similar:\n",
+    "        return [scotch]\n",
+    "    else:\n",
+    "        return [scotch, similar[0]]"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "urth": {
+     "dashboard": {
+      "hidden": true
+     }
+    }
+   },
+   "source": [
+    "## Dashboard Setup <span style=\"float: right; font-size: 0.5em\"><a href=\"#Got-Scotch?\">Top</a></span>"
```

Comment:
Probably want to call this Widget Setup. Dashboard layout is waaayyyy at the bottom.